godaddy-wordpress/coblocks
# The Lesser Migration

### Description


The Great Migration POC demonstrated a successful approach for migrating CoBlocks blocks into Core blocks, providing users with a more consistent and unified experience. The Lesser Migration builds on this success by implementing a staged approach, allowing us to gradually introduce this experience starting with a smaller audience.

As the Gutenberg landscape has evolved, it has become clear that Core blocks offer quality user experience, consistent controls, and regular updates. While it is possible to align CoBlocks with Core, it has proven to be a challenging task to back-port features and functionalities. Additionally, migrations to new features and functionality often require deprecating blocks, adding to bloat and load times.

In summary, The Lesser Migration aims to streamline the user experience by migrating select blocks to Core, reducing maintenance and update overhead, and allowing us to focus on unique and user-enhancing features in CoBlocks.

Also requires a Go branch for proper Go-specific styling.
https://github.com/godaddy-wordpress/go/pull/838

### Screenshots


**Alert Block**
![migrateAlertBlock](https://user-images.githubusercontent.com/30462574/215218208-02c9d293-4190-460a-b931-a061c6df53e8.gif)

**Author Block**
![migrateAuthorBlock](https://user-images.githubusercontent.com/30462574/215218264-3953bd62-6779-4cab-9bbb-f3d3aff52f7e.gif)

### Types of changes




Major changes intended for CoBlocks 3.0 release.
PHP and JS changes introducing block migrations. 


### How has this been tested?

Manually tested minor CoBlocks versions back to 2.20.0 all able to successfully migrate.
PHPUnit tests in place. Unit tests cover:
- [x] functionality of core migration logic.
- [x] block parser attribute validation 
- [x] e2e test covering JavaScript logic of migration - Handled in Cypress in GHA.

**To test this manually:**
First, you must set up a 'master' branch of CoBlocks locally. You want to create two new posts. One post is to test the Author block and another is to test the Alert block. When running the block editor and 'master' branch you should insert blocks into the post. We should use features of the block as well. For example, we should have a block with custom colors. A block with custom alignment. Block with a custom class name. A Block using any and all features such as block animation or typography controls like bold and italics. 

Once we have a test post for each block, and each test post is filled with configured blocks, we are ready to test!

Next, we need to check out this branch `cb-30-base` and build. After building these changes, the next time we load one of the test posts, it will migrate the CoBlocks blocks into Core blocks.


For reference:
Alert Block -> Paragraph styled as an alert.
Author Block -> Columns block layout as author. 

### Acceptance criteria



- [x] Blocks should migrate. 
- [x] Tests should provide confidence of functionality.

### Checklist:
